src/workbench/SideBarHeader.tsx

```diff
@@ -66,16 +68,19 @@ const SideBarHeader = () => {
       debounce(async (newQuery: string) => {
         const trimmedQuery = newQuery.trim();
         if (trimmedQuery) {
-          const results = await search.search(trimmedQuery);
+          const newResults = await search.search(trimmedQuery);
           if (!isUnmounted()) {
-            setResults(results);
+            if (!results) {
+              logging.event({ type: "search" });
+            }
+            setResults(newResults);
           }
         } else {
           setResults(undefined);
         }
         setViewedResults([]);
       }, 300),
-    [search, setResults, setViewedResults, isUnmounted]
+    [search, setResults, setViewedResults, isUnmounted, logging, results]
```

Comment:
Can we use the same trick as we discussed on zoom to avoid depending on results here by passing a function to setResults? Nothing else we depend on is expected to change.

Reply:
Please see [`a2b0e40`](https://github.com/microbit-foundation/python-editor-next/pull/519/commits/a2b0e40ef71c0079052dc2d5e10cdc66ff3c57e5).